### PR TITLE
Migrate `autoConfirmFingerPrints` to `StateProvider`

### PR DESCRIPTION
**Pull request by addisonbeck**
# Migrate `autoConfirmFingerPrints` to `StateProvider`

## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This PR migrates the `autoConfirmFingerPrints` data point from `StateService`
to `StateProvider`.

## Code changes

`autoConfirmFingerPrints` is a bit field in Bitwarden's client side state. It
is set to control the appearance (or lack of appearance) of a fingerprint
confirmation dialogue during two common workflows: confirming a user to an
organization, and confirming a key-holder for emergency access. 

Currently this data point is fetched from `StateService` directly in
components with no intermediary stateful service. This is not compliant with
the design of `StateProvider`, and so the first step in this work was to wrap
a service around the concept of UX preferences for organization management,
and to update components to reference this service instead.

The work can (hopefully) be logically followed by commit. There is one commit
that modifies some Auth team files. Auth owns the emergency access feature,
and the emergency access feature was impacted by the new service being
created. There is no logical change here and this should just be a refactor.

You may be asking yourself "why does emergency access confirmation and
organization member confirmation reference the same value for this setting?"
Well, me too. This has been logged as a bit of tech debt for Admin Console to
split up, and a link to that ticket can be found below if desired.

I also fixed a typo in the `StateDefinition` documentation. This is unrealted
to the work, but I was in the area and figured nobody would mind.

## References

* [Jira work item for this work AC-2011](https://bitwarden.atlassian.net/browse/AC-2011)
* [Jira work item to separate the two `autoConfirmFingerPrints`
options](https://bitwarden.atlassian.net/browse/AC-2315)

### CODE REVIEW

**Review by MGibson1 — CHANGES_REQUESTED** [review]
Looks good, just a few variable name nits. The service parameter name is one that could potentially be annoying down the road with repo searches

File: libs/common/src/admin-console/abstractions/organization-management-preferences/organization-management-preferences.service.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Observable } from "rxjs";
+
+/**
+ * Manages the state of a single organization management preference.
+ * Can be used to subscribe to or update a given property.
+ */
+export class OrganizationManagementPreference<T> {
```
[the remark this answers is not in the document]
Comment (addisonbeck) [reply libs/common/src/admin-console/abstractions/organization-management-preferences/organization-management-preferences.service.ts:7]: ✅ Implement  as suggested on https://github.com/bitwarden/clients/pull/8337/commits/e6eeb2b7545654630c37c21b5e9199c8c3a94d48

File: apps/web/src/app/admin-console/organizations/manage/user-confirm.component.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ export class UserConfirmComponent implements OnInit {
     }
 
     if (this.dontAskAgain) {
-      await this.stateService.setAutoConfirmFingerprints(true);
+      await this.organizationManagementPreferenceService.autoConfirmFingerPrints.update(true);
```
[the remark this answers is not in the document]
Comment (addisonbeck) [reply apps/web/src/app/admin-console/organizations/manage/user-confirm.component.ts]: ✅ Implemented from suggestion on https://github.com/bitwarden/clients/pull/8337/commits/d426201777eb1967e57d7b9275fcdac2ec7a031e

File: apps/web/src/app/admin-console/organizations/manage/user-confirm.component.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export class UserConfirmComponent implements OnInit {
   constructor(
     private cryptoService: CryptoService,
     private logService: LogService,
-    private stateService: StateService,
+    private organizationManagementPreferenceService: OrganizationManagementPreferencesService,
```
[the remark this answers is not in the document]
Comment (addisonbeck) [reply apps/web/src/app/admin-console/organizations/manage/user-confirm.component.ts]: ✅ Implemented from suggestion on https://github.com/bitwarden/clients/pull/8337/commits/9480c27c7874417ef985d1fefc652c05a186b2d6

**Review by MGibson1 — DISMISSED** [review]

**Review by eliykat — DISMISSED** [review]
Cool way of managing disparate pieces of state through a single service 👏

**Review by ike-kottlowski — DISMISSED** [review]
Auth changes are good.

**Review by ike-kottlowski — APPROVED** [review]

**Review by eliykat — APPROVED** [review]

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**